src/Lib/Canvas.vala

```diff
@@ -493,6 +496,20 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         }
     }
 
+    /*
+     * Will update snap decorators if necessary.
+     */
+    private void on_update_snap_decorators () {
+        var extra_context = mode_manager.active_mode_extra_context ();
+        if (extra_context is Akira.Lib.Modes.TransformMode.TransformExtraContext) {
+            var tec = extra_context as Akira.Lib.Modes.TransformMode.TransformExtraContext;
+            snap_manager.generate_decorators (tec.snap_guide_data);
+        }
+        else if (snap_manager.is_active ()) {
+            snap_manager.reset_decorators ();
+        }
+    }
```

Comment:
nit code styling.

The else conditions is not really an alternative to the active extra context, so stylistically I prefer an early return and then a following condition. This gives a good sense of the priority in which we want to check for things.

Reply:
The tacit else is: Nothing wants snap guides, and there isn't an active one to reset; so don't do anything. I don't think an early return is the right fix here.